admin/linux/build-appimage.sh

```diff
@@ -62,11 +65,11 @@ rm -rf ./usr/share/caja-python/
 rm -rf ./usr/share/nautilus-python/
 rm -rf ./usr/share/nemo-python/
 
-# Move sync exlucde to right location
+# Move sync exclude to right location
 mv ./etc/Nextcloud/sync-exclude.lst ./usr/bin/
 rm -rf ./etc
 
-sed -i -e 's|Icon=nextcloud|Icon=Nextcloud|g' usr/share/applications/nextcloud.desktop # Bug in desktop file?
+sed -i -e 's|Icon=nextcloud|Icon=Nextcloud|g' usr/share/applications/${LINUX_APPLICATION_ID}.desktop # Bug in desktop file?
```

Comment:
it's a bit surprising to me that the wrongly named icon seems to work.
IIRC the name of the icon is supposed to be the app's id, i.e. `share/icons/hicolor/128x128/apps/com.nextcloud.desktopclient.nextcloud.png`.  In any case, naming the icon after the appid is certainly more semantically correct than claiming any random name.
Also note that  `desktop-file-edit --set-icon=ICON` exists.

Reply:
Indeed it works, but I'm not that surprised - after glimpsing through
https://fedoraproject.org/wiki/Workstation/softwareappdata
https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#introduction
https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
the full reverse-DNS naming scheme has nothing to do with the application icons which should just be named with their simple appname (or specific names when integrating them like appname-highcontrast-48px or whatever) and then be specified in the desktopfile with Icon=AppName. The last link above applies to themes files where only short name is used too.

However, If this is outdated or I misunderstood something I'd be happy to learn something new.

Reply:
Hi @ivaradi, would you know more about it? Thanks.

Reply:
@DominiqueFuchs most apps i have seen do use the same name for the desktop file and the icons.
is there any reason not to change the icons to the reverse-DNS scheme?
It is a requirement for Flatpak apps (http://docs.flatpak.org/en/latest/conventions.html#application-icons).
And it would allow me to remove the [manual renaming](https://github.com/flathub/org.nextcloud.Nextcloud/blob/master/org.nextcloud.Nextcloud.yml#L10) that is currently required for the client on flathub.